npc/events/christmas_2006.txt

```diff
@@ -0,0 +1,4165 @@
+				.@mobName$ = "Bon Gun";
+				.@delItem = 659; // Heart_Of_Her
+				.@cakeAmount = 7;
+				.@cookieAmount = 3;
+				break;
+			case 21:
+				mes "["+ strcharinfo(0) +"]";
+				mes "Here!!I've prepared an 'Horror of tribe'! You can get to know about the strict rule of the tribe with it. Orc Warrior likes it!";
+				.@mobName$ = "Orc Warrior";
+				.@delItem = 635; // Horror_Of_Tribe
+				.@cakeAmount = 7;
+				.@cookieAmount = 3;
+				break;
+			case 22:
+				mes "["+ strcharinfo(0) +"]";
+				mes "Here!!I've prepared an 'Book of devil'!A forbidden book in which the secret for summoning high level demons is detailed. Bapho Jr likes it!";
+				.@mobName$ = "Bapho Jr";
+				.@delItem = 642; // Book_Of_Devil
+				.@cakeAmount = 10;
+				.@cookieAmount = 3;
+				break;
+			case 23:
+				mes "["+ strcharinfo(0) +"]";
+				mes "Here!!I've prepared an a 'contract in the dark'! It's a contract as it's called. Deviruchi likes it!";
+				.@mobName$ = "Deviruchi";
+				.@delItem = 641; // Contracts_In_Shadow
+				.@cakeAmount = 10;
+				.@cookieAmount = 3;
+				break;
+			case 24:
+				mes "["+ strcharinfo(0) +"]";
+				mes "Here!!I've prepared an 'silver knife of the purity'! It represents the virginity. Sohee likes it!";
+				.@mobName$ = "Sohee";
+				.@delItem = 638; // Silver_Knife_Of_Chaste
+				.@cakeAmount = 10;
+				.@cookieAmount = 3;
+				break;
+			case 25:
+				mes "["+ strcharinfo(0) +"]";
+				mes "Here!!I've prepared an 'tropical banana'! A premium quality banana grown in the tropics. Yoyo likes it!";
+				.@mobName$ = "Yoyo";
+				.@delItem = 634; // Tropical_Banana
+				.@cakeAmount = 10;
+				.@cookieAmount = 3;
+				break;
+		}
+		next;
+		mes "[Poe]";
+		mes "Oh... you mean I can tame "+.@mobName$+" with this?";
+		mes "Thank you. Please take good care of my brother.";
+		next;
+		if (countitem(.@delItem) > 0) {
+			mes "[Poe]";
+			mes "Breed him well! Oh, this is a Christmas gift for you! Merry Christmas I appreciate your kindness!";
+			delitem .@delItem, 1;
+			06christ_santa = 5;
+			getitem 12200, .@cakeAmount; // Event_Cake
+			getitem 12130, .@cookieAmount; // Cookie_Bag
+			makepet 1245; // GOBLINE_XMAS
+			close;
+		} else {
+			mes "[Poe]";
+			mes "So, where's it?";
+			next;
+			mes "["+ strcharinfo(0) +"]";
+			mes "Hum, here it is!";
+			mes " ";
+			mes "...um..it must be inside...the..um";
+			next;
+			mes "["+ strcharinfo(0) +"]";
+			mes "Dear! Where has it gone?";
+			mes "Hang on, give me some time to look for it!";
+			close;
+		}
+	} else {
+		mes "[Poe]";
+		mes "However, I wanted to breed it. I guess it should be difficult to the little boy like me.";
+		next;
+		mes "[Poe]";
+		mes "I wish you to breed it instead of me. You said you were free but I'd like to say 'Thank you' for your help.";
+		close;
+	}
+}
+
+prontera,226,306,0	script	Ohholy#06ohholy_01	1_F_PRIEST,{
+	if (06christ_santa < 1) {
```

Comment:
`.@name$ = strcharinfo(0);` to avoid the redundant command call

Reply:
Applied suggested fix.